modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/ExtensionDefinitionParser.java

```diff
@@ -513,89 +515,106 @@ protected ClassLoader getContextClassLoader() {
 
     final Class<Object> expectedClass = getType(expectedType);
 
-    if (isExpressionFunction(modelProperties) && value != null) {
-      resolver = new ExpressionFunctionValueResolver<>((String) value, expectedType, muleContext);
-    } else if (isParameterResolver(modelProperties) && value != null) {
-      resolver = new ExpressionBasedParameterResolverValueResolver<>((String) value, expectedType, muleContext);
-    } else if (isTypedValue(modelProperties)) {
-      resolver = new ExpressionTypedValueValueResolver((String) value, expectedClass, muleContext);
+    boolean isExpression = isExpression(value, parser);
+
+    if (isExpression) {
+      if (isExpressionFunction(modelProperties)) {
+        resolver = new ExpressionFunctionValueResolver<>((String) value, expectedType, muleContext);
+      } else if (isParameterResolver(modelProperties, expectedType)) {
```

Comment:
this doesn't make sense... you already know if this is an expression or not. You shouldn't test it for parameter resolver twice (here and in line 539 just for selecting different resolvers). Use the isExpression variable in one and only if.